pkg/reconciler/channel/channel.go

```diff
@@ -102,22 +104,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 	// Reconcile this copy of the Channel and then write back any status
 	// updates regardless of whether the reconcile error out.
-	err = r.reconcile(ctx, channel)
-	if err != nil {
-		logging.FromContext(ctx).Warn("Error reconciling Channel", zap.Error(err))
+	reconcileErr := r.reconcile(ctx, channel)
```

Comment:
serving does not do this: https://github.com/knative/serving/blob/master/pkg/reconciler/revision/revision.go#L238-L253

Reply:
We do some places: https://github.com/knative/serving/blob/dbdd4ca51c8c5b77ef926c0bb48f663ccbcb1c64/pkg/reconciler/revision/reconcile_resources.go#L114

Reply:
Changed serving and build to follow this pattern.